src/main/java/Duke.java

```diff
@@ -1,10 +1,447 @@
+                throw new Exception("Error type");
+            }
+        }
+    }
+    /**
+     * Generate the initiate message.
+     */
+    private void greet() {
+            System.out.println(line);
+            String logo = " ____        _        \n"
+                    + "|  _ \\ _   _| | _____ \n"
+                    + "| | | | | | | |/ / _ \\\n"
+                    + "| |_| | |_| |   <  __/\n"
+                    + "|____/ \\__,_|_|\\_\\___|\n";
+            System.out.println(logo);
+            System.out.println("Hello! I'm Duke\n");
+            System.out.println("What can I do for you?");
+            System.out.println(line);
+        }
+
+    /**
+     * generate the end message.
+     */
+    private void bye() {
+        String line = "__________________________________";
+        System.out.println("Bye. Hope to see you again soon!");
+        System.out.println(line);
+        saveListInFile("src/main/java/Duke.txt");
+    }
+    /**
+     * The original method to simply generate an echo message.
+     * Now only meant to decide if the user has decided to end
+     * the program.
+     * @return boolean
+     */
+    private boolean echo() {
+        Scanner user = new Scanner(System.in);
+        String input = user.nextLine();
+        if (input.equals("bye")) {
+            bye();
+            return false;
+        } else {
+            return choiceOfAction(input);
+        }
+    }
+
+    /**
+     * function adds a new Task into the taskList if valid.
+     * @param newTask
+     */
+    private void addTask(Task newTask) {
+        taskList.add(newTask);
+        System.out.println(line);
+        System.out.println("Got it. I've added this task: ");
+        System.out.println("    " + newTask);
+        System.out.format("Now you have %d tasks in the list.\n", taskList.size());
+    }
+
+    /**
+     * function removes the task located at the index of the task list.
+     * @param newInput
+     */
+    private void deleteTask(String[] newInput ) throws Exception{
+
+        if (newInput.length < 2) {
+            throw new Exception("invalid Syntax for delete instruction");
+        } else {
+            int index = Integer.parseInt(newInput[1]);
+            if (index > 0 && index <= taskList.size()) {
+                System.out.println("Noted. I've removed this task: ");
+                System.out.println("    " + taskList.get(index - 1).toString());
+                taskList.remove(index - 1);
+                System.out.format("Now you have %d tasks in the list.\n", taskList.size());
+            } else {
+                throw new Exception("index not within valid range");
+            }
+        }
+    }
+
+    /**
+     * To decide base on the input what is the next action.
+     * The decision on whether to show list, set existing task to
+     * done, or to create and record new task.
+     * @param input
+     * @return boolean
+     */
+    private boolean choiceOfAction(String input) {
```

Comment:
Since this is a function that returns a boolean result, I do feel the naming should be isSetDone or isNewTask.